src/java/org/apache/cassandra/cql3/Operator.java

```diff
@@ -44,6 +58,34 @@ public boolean isSatisfiedBy(AbstractType<?> type, ByteBuffer leftOperand, ByteB
         {
             return type.compareForCQL(leftOperand, rightOperand) == 0;
         }
+
+        @Override
+        public boolean requiresFilteringOrIndexingFor(ColumnMetadata.Kind columnKind)
+        {
+            return !columnKind.isPrimaryKeyKind();
+        }
+
+        @Override
+        public RangeSet<ClusteringElements> restrict(RangeSet<ClusteringElements> rangeSet, List<ClusteringElements> args)
+        {
+            assert args.size() == 1;
```

Comment:
nit: Given we never run w/ `-da`, perhaps we could add a short error message (I'm guessing saying we can't equal multiple things at the same time)?